File: src/main/scala/com/tattyseal/hgp/block/ItemBlockStainedBlock.scala
package com.tattyseal.hgp.block

import net.minecraft.block.Block
import net.minecraft.init.{Blocks, Items}
import net.minecraft.item.{ItemStack, ItemBlock}

/**
 * Created by Toby on 05/02/2015.
 */
class ItemBlockStainedBlock(block: Block) extends ItemBlock(block)
{
  setHasSubtypes(true);
  setUnlocalizedName("hgp.coloredStainedPane");

  override def getUnlocalizedName(stack: ItemStack): String =
  {
    return super.getUnlocalizedName() + "." + stack.getItemDamage();
  }

  override def getMetadata(damage: Int): Int =
  {
    return damage;
  }

  override def getItemStackDisplayName(stack: ItemStack): String = new ItemStack(Blocks.stained_glass_pane, 1, stack.getItemDamage()).getDisplayName();
}

File: src/main/java/com/tattyseal/hgp/item/ItemBlockStainedBlock.java
package com.tattyseal.hgp.item;

import com.tattyseal.hgp.HorizontalGlassPanes;
import net.minecraft.block.Block;
import net.minecraft.init.Blocks;
import net.minecraft.item.ItemBlock;
import net.minecraft.item.ItemStack;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;

public class ItemBlockStainedBlock extends ItemBlock
{
	public ItemBlockStainedBlock(Block b)
	{
		super(b);
		setHasSubtypes(true);
		setUnlocalizedName("hgp.coloredStainedPane");
	}
	
	@Override
	public String getUnlocalizedName(ItemStack stack) 
	{
		return super.getUnlocalizedName() + "." + stack.getItemDamage();
	}
	
	@Override
	public int getMetadata(int damage) 
	{
		return damage;
	}

	@SideOnly(Side.CLIENT)
	@Override
	public String getItemStackDisplayName(ItemStack stack) 
	{
		return HorizontalGlassPanes.proxy.getHorizontal() + " " + new ItemStack(Blocks.STAINED_GLASS_PANE, 1, stack.getItemDamage()).getDisplayName();
	}
}
